x-pack/plugins/actions/server/builtin_action_types/cases_webhook/utils.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AxiosResponse, AxiosError } from 'axios';
+import { isEmpty, isObjectLike, get } from 'lodash';
+import { addTimeZoneToDate, getErrorMessage } from '../lib/axios_utils';
+import * as i18n from './translations';
+
+export const createServiceError = (error: AxiosError, message: string) =>
+  new Error(
+    getErrorMessage(
+      i18n.NAME,
+      `${message}. Error: ${error.message}. ${
+        error.response?.statusText != null ? `Reason: ${error.response?.statusText}` : ''
+      }`
+    )
+  );
+
+export const getPushedDate = (timestamp?: string) => {
+  if (timestamp != null && new Date(timestamp).getTime() > 0) {
+    try {
+      return new Date(timestamp).toISOString();
+    } catch (e) {
+      return new Date(addTimeZoneToDate(timestamp)).toISOString();
+    }
+  }
+  return new Date().toISOString();
+};
+
+export const getObjectValueByKey = <T = unknown>(
+  obj: Record<string, Record<string, T> | T>,
+  key: string
+): T => {
+  // @ts-ignore
+  return get(obj, key);
+};
+
+export const throwIfResponseIsNotValidSpecial = ({
+  res,
+  requiredAttributesToBeInTheResponse = [],
+}: {
+  res: AxiosResponse;
+  requiredAttributesToBeInTheResponse?: string[];
+}) => {
+  const requiredContentType = 'application/json';
+  const contentType = res.headers['content-type'];
+  const data = res.data;
+
+  /**
+   * Check that the content-type of the response is application/json.
+   * Then includes is added because the header can be application/json;charset=UTF-8.
+   */
+  if (contentType == null) {
+    throw new Error(
+      `Missing content type header in ${res.config.method} ${res.config.url}. Supported content types: ${requiredContentType}`
+    );
+  }
+  if (!contentType.includes(requiredContentType)) {
+    throw new Error(
+      `Unsupported content type: ${contentType} in ${res.config.method} ${res.config.url}. Supported content types: ${requiredContentType}`
+    );
+  }
+
+  /**
```

Comment:
Maybe remove this comment, I don't think it matches the logic in the `if` block.